sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java

```diff
@@ -461,10 +451,11 @@ private IAccount retrieveAccount(String id) {
 
     @WorkerThread
     private synchronized void selectSignInTypeAndSignIn() {
+        if (!NetworkStateHelper.getSharedInstance(mContext).isNetworkConnected()) {
+            completeSignIn(null, new NetworkErrorException("Sign-in failed. No internet connection."));
```

Comment:
I'm not sure, do we need a `return` after `completeSignIn`?

Reply:
@guperrot does completing the future stops any further execution?

Reply:
@annakocheshkova no

Reply:
it just releases the thread that was calling `get()` or post a runnable into ui thread if using `thenAccept` but this thread is the worker thread and just continue running after complete and you need to return.